File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.cs
public class Solution : GuessGame {
    public int GuessNumber(int n) {
        int left = 1;
        int right = n;
        while(left <= right){
            int mid = left+(right-left)/2;
            int result = guess(mid);
            if (result == 1)
            left = mid + 1;
            else if ( result == 0)
            return mid;
            else
            right = mid - 1;
        }
        return -1;
    }
}

File: 374-guess-number-higher-or-lower/guess-number-higher-or-lower.js
var guessNumber = function(n) {
    let left = 1, right = n;
    while(left <= right){
        let mid = Math.floor((right + left)/2);
        let result = guess(mid);
        if (result == -1)
            right = mid - 1;
        else if(result == 1)
            left = mid + 1;
        else
        return mid;
    }
    
};